src/transcript/keccak.rs

```diff
@@ -0,0 +1,101 @@
+use std::marker::PhantomData;
+use tiny_keccak::{Keccak, Hasher};
+use ark_ec::CurveGroup;
+use ark_ff::{BigInteger, PrimeField};
+
+use crate::transcript::Transcript;
+
+/// KecccakTranscript implements the Transcript trait using the Keccak hash
+pub struct KeccakTranscript<C: CurveGroup> {
+    sponge: Keccak,
+    phantom: PhantomData<C>,
+}
+
+#[derive(Debug)]
+pub struct KeccakConfig {}
+
+impl<C: CurveGroup> Transcript<C> for KeccakTranscript<C> {
+    type TranscriptConfig = KeccakConfig;
+    fn new(config: &Self::TranscriptConfig) -> Self {
+        let _ = config;
+        let sponge = Keccak::v256();
+        Self {
+            sponge,
+            phantom: PhantomData,
+        }
+    }
+
+    fn absorb(&mut self, v: &C::ScalarField) {
+        self.sponge.update(&(v.into_bigint().to_bytes_le()));
+    }
+    fn absorb_vec(&mut self, v: &[C::ScalarField]) {
+        for _v in v {
+            self.sponge.update(&(_v.into_bigint().to_bytes_le()));
+        }
+    }
+    fn absorb_point(&mut self, p: &C) {
+        let mut serialized = vec![];
+        p.serialize_compressed(&mut serialized).unwrap();
+        self.sponge.update(&(serialized))
```

Comment:
Regarding simulating the transcript inside a SNARK, I currently think that we will not need it for the Keccak transcript as mentioned in [this comment](https://github.com/privacy-scaling-explorations/folding-schemes/issues/14#issuecomment-1721306866), since the keccak transcript is intended to be used only in non-circuit cases (eg. in Nova's main transcript), but not in-circuit (eg. HyperNova's SumCheck transcript, where we would use for example Poseidon's Transcript to later reproduce it in-circuit) (more details in the [original comment itself](https://github.com/privacy-scaling-explorations/folding-schemes/issues/14#issuecomment-1721306866)).